evennia/contrib/random_string_generator.py

```diff
@@ -0,0 +1,340 @@
+"""
+Pseudo-random generator and registry
+
+Evennia contribution - Vincent Le Goff 2017
+
+This contrib can be used to generate pseudo-random strings of information
+with specific criteria.  You could, for instance, use it to generate
+phone numbers, license plate numbers, validation codes, non-sensivite
+passwords and so on.  The strings generated by the generator will be
+stored and won't be available again in order to avoid repetition.
+Here's a very simple example:
+
+```python
+from evennia.contrib.random_string_generator import Generator
+# Create a generator for phone numbers
+phone_generator = Generator("phone number", r"555-\d{3}-\d{4}")
+# Generate a phone number (555-XXX-XXXX with X as numbers)
+number = phone_generator.get()
+# `number` will contain something like: "555-981-2207"
+# If you call `phone_generator.get`, it won't give the same anymore.
+phone_generator.all()
+# Will return a list of all currently-used phone numbers
+phone_generator.remove("555-981-2207")
+# The number can be generated again
+```
+
+To use it, you will need to:
+
+1. Import the `Generator` class from the contrib.
+2. Create an instance of this class taking two arguments:
+   - The name of tje gemerator (like "phone number", "license plate"...).
+   - The regular expression representing the expected results.
```

Comment:
Here it could be good to give a link to the Python regex documentation: `https://docs.python.org/2/library/re.html`